overlord/servicestate/service_control_test.go

```diff
@@ -773,6 +782,66 @@ func (s *serviceControlSuite) TestRestartServices(c *C) {
 	})
 }
 
+func (s *serviceControlSuite) TestRestartWithExplicitServices(c *C) {
+	st := s.state
+	st.Lock()
```

Comment:
this slightly suprisingly seems not create trouble, but is anyway a good habit for the production code to follow this with a defer st.Unlock() as well

Reply:
agreed, I think this should become:

```suggestion
	st.Lock()
	defer st.Unlock()
```

Reply:
Done